### DISCUSSION TITLE
Question about blog as form widget

### DISCUSSION THREAD

**Discussion by Zmove** [OFFLINE-GmbH/oc-boxes-support#143] [Help]
Hello,

I'm using boxes as a form widget to manage a blog content. On the frontend, it works well, by following the doc, I can see my boxes in my content. This is the last block with text on the bottom of my screenshot.

![image](https://github.com/OFFLINE-GmbH/oc-boxes-support/assets/1314678/306a0030-a919-46ea-957d-960deda90a3b)

But on the backend, when I'm editing my content, all my post header is empty. I don't see how can I pass the slug to get access to the current record to be able to display all associated data (title etc....)

![image](https://github.com/OFFLINE-GmbH/oc-boxes-support/assets/1314678/1c324c46-0367-46c0-8514-36dbb377d4a4)

Thank you for your help

**Comment by tobias-kuendig** [comment]
You need to share some code and details about your setup. How does your `author` box find out what to display?

Usually, it is done like so:

https://docs.boxes.offline.ch/use-cases/usage-in-plugins.html#get-the-related-model-from-a-partial

**Reply by Zmove** [reply]
@tobias-kuendig thank you for your answer.

Here is my setup.

post.yaml

    fields:
        boxes:
            label: Boxes
            tab: Content
            span: adaptive
            type: boxes
            previewLayout: post

Here is my layout (just put the important part)

    [section category]
    handle = "Blog\Categories"
    identifier = "fullslug"
    value = "{{ :fullslug }}"
    isDefault = 0
    
    [collection categories]
    handle = "Blog\Categories"
    
    [collection posts]
    handle = "Blog\Posts"
    
    [section record]
    handle = "Blog\Posts"
    identifier = "slug"
    value = "{{ slug }}"
    isDefault = 1

    ......
    <article id="post">
        <a class="badge" href="{{ 'blog/records'|page({fullslug:record.category.fullslug}) }}">{{ record.category.title }}</a>
        <h1 class="mt-sm mb-md title">{{ record.title }}</h1>
        <footer class="meta row g-lg align-center text-lightgrey font-md">
            <div class="author row g-sm align-center">
                {% partial 'image' class="radius-50" image=record.author.image alt=record.author.title width=36 height=36 %}
                {{ record.author.title }}
            </div>
            <time class="date" datetime="{{ record.published_at|date('Y-m-d') }} }}">
                {{ carbon(record.published_at).format('d F Y') }}
            </time>
        </footer>
        <div class="image my-md">
            {% partial 'image' image=record.image|media alt=record.title width=800 height=460 class="radius-sm w-100" %}
        </div>
        <div class="content">
            {% page %}
        </div>
    </article>
    ......

Here is my page 

    url = "/article/:fullslug*/:slug"
    layout = "post"
    title = "Détail d'un article"
    ==
    {% if record is empty %}
        {% do abort(404) %}
    {% endif %}
    
    {{ record.boxes.render()|raw }}

Thank you for your help

**Reply by tobias-kuendig** [reply]
You are fetching your record using `value = "{{ slug }}"` but this cannot work in the backend, since there is no slug (the preview just renders your layout with your Boxes, but it never knows on what page you will render it, so there is no slug).

You could use [Page Templates](https://docs.boxes.offline.ch/use-cases/page-templates.html) and add your intro/image section of the blog layout as a Boxes (use `locked: true` so the user cannot edit them).

Another option is to use `this.page.apiBag['boxes-is-editor']` in your layout. If it is true, the layout is rendered in the Editor, just display dummy data for your author/published date.

**Reply by PubliAlex** [reply]
> You are fetching your record using value = "{{ slug }}" but this cannot work in the backend, since there is no slug (the preview just renders your layout with your Boxes, but it never knows on what page you will render it, so there is no slug).

This is what I thought, but I would suppose there was a solution to this bundled with boxes module as it's a very common case as soon as you work with boxes as form widget for tailor.

> You could use [Page Templates](https://docs.boxes.offline.ch/use-cases/page-templates.html) and add your intro/image section of the blog layout as a Boxes (use locked: true so the user cannot edit them).

It's not a solution in my case because that page have sidebars where I display some informations that are not in the scope of boxes. and that sidebars also depend on the slug.

Normally, the slug can be known as I'm on the edit form of the current post. We just have to fill that info when we create the boxes form widget to provide extra informations about the variable in the page url pattern, and the fields containing the value.

    fields:
        boxes:
            label: Boxes
            tab: Content
            span: adaptive
            type: boxes
            previewLayout: post
            parameters:
                slug: slug # url_param: model_field_name


> Another option is to use this.page.apiBag['boxes-is-editor'] in your layout. If it is true, the layout is rendered in the Editor, just display dummy data for your author/published date.

Not ideal too, the goal of the box is to see what you get. If it's to see a page with some dummy datas, it fails as it's goal.

Thank you
